airflow/contrib/operators/kubernetes_pod_operator.py

```diff
@@ -158,17 +167,17 @@ def execute(self, context):
                 raise AirflowException(
                     'Pod returned a failure: {state}'.format(state=final_state)
                 )
-            if self.xcom_push:
+            if self.do_xcom_push:
                 return result
         except AirflowException as ex:
             raise AirflowException('Pod Launching failed: {error}'.format(error=ex))
 
     def _set_resources(self, resources):
-        inputResource = Resources()
-        if resources:
-            for item in resources.keys():
-                setattr(inputResource, item, resources[item])
-        return inputResource
+        return Resources(**resources) if resources else Resources()
+
+    def _set_name(self, name):
+        validate_key(name, max_length=63)
```

Comment:
Hi,

I've just upgraded from v.1.10.6 and thus [this change has been included](https://airflow.apache.org/docs/stable/changelog.html#airflow-1-10-7-2019-12-24), breaking a lot of the DAGs we currently have setup due to this change.
**I hugely appreciate the effort in this development**, although fair warning of this _breaking change_ would have been very appreciated.

Just leaving this here in case someone using `KubernetesPodOperator` finds the same _issue_ when upgrading.
```python
...
  File "/usr/local/lib/python3.6/site-packages/airflow/contrib/operators/kubernetes_pod_operator.py", line 167, in __init__ 
    self.name = self._set_name(name) 
  File "/usr/local/lib/python3.6/site-packages/airflow/contrib/operators/kubernetes_pod_operator.py", line 269, in _set_name 
    validate_key(name, max_length=63) 
  File "/usr/local/lib/python3.6/site-packages/airflow/utils/helpers.py", line 64, in validate_key 
    "The key has to be less than {0} characters".format(max_length)) 
airflow.exceptions.AirflowException: The key has to be less than 63 characters
```

I understand [Kubernetes has a character limit of 63 for its label's names](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) but, at least in v1.10.6 (the latest version previous to this change) [Airflow adds another 8 characters to the name](https://github.com/apache/airflow/blob/1.10.6/airflow/contrib/kubernetes/pod_generator.py#L167), so I'm guessing the limit to set here would be `55`, right?

Please let me know if I'm missing something.

Again, thanks a lot for the work on this, I mean this as constructive feedback.

Reply:
Interesting that it's a breaking change, I was in the understanding that it would fail downstream and hence was a good check to move upstream to point of DAG creation. If this causes errors that didn't occur before, maybe it should be loosened (what do you say @potiuk) and for sure mentioned in UPDATING.md.

Reply:
Sorry, I'm not an expert in Kubernetes.

I was creating the POD names in `KubernetesPodOperator` just concatenating the DAG name with the task name and haven't had any issue whatsoever. Suddenly when updating Airflow I started to see how many of my DAGs wouldn't be working as they didn't pass the newly-added validation.

Might we be confusing the limit on the _labels_ (`63` characters) and the _subdomains_ (`253` characters)?

I cannot get a clear view reading the docs:
- [Object Names and IDs](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/)
- [Identifiers and Names in Kubernetes](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md)

Also the [_unofficial Kubernetes_ mentions a limit of `253` characters](https://unofficial-kubernetes.readthedocs.io/en/latest/concepts/overview/working-with-objects/names/).

Reply:
Hey @ddelange, I can confirm that the limit for the POD name is **`253` characters** and not `63`.

Tried out with one of my colleagues in our _k8s_ cluster, only giving an error when trying to create a POD with more than `253` characters:

```sh
The Pod "eot5ahm9ua4ocahmahghoma1xoh7neihuvohz3oofaiz1iesaet2eibuwee5didietheet7faelohceyo1ci2eithu9nutie5fee5ahhohy0haegaikeizohngahzahteseifo5au2eb0aizi7reph9eibo5sahlee7hiequ2aeko6yiam8ieca4si6hodeiquoh6ceu9ienge2ooh4uo2umaijaec4aeli4ohfeodie3xahkove2iogieno2i" is invalid: metadata.name: Invalid value: "eot5ahm9ua4ocahmahghoma1xoh7neihuvohz3oofaiz1iesaet2eibuwee5didietheet7faelohceyo1ci2eithu9nutie5fee5ahhohy0haegaikeizohngahzahteseifo5au2eb0aizi7reph9eibo5sahlee7hiequ2aeko6yiam8ieca4si6hodeiquoh6ceu9ienge2ooh4uo2umaijaec4aeli4ohfeodie3xahkove2iogieno2i": must be no more than 253 characters
```